lms/djangoapps/completion/utils.py

```diff
@@ -0,0 +1,166 @@
+
+    def get_activities(self, user):
+        """
+        Return the avalaible activities on the course
+        """
+        usage_key = modulestore().make_course_usage_key(self.course_key)
+        blocks = get_course_blocks(user, usage_key)
+
+        block_types_filter = [
+            'html',
+            'problem',
+            'video',
+            'discussion',
+        ]
+
+        if self.site_name is not None:
+            try:
+                site = Site.objects.get(name=self.site_name)
+                custom_block_types = site.microsite.values.get(CUSTOM_BLOCK_TYPES)
+
+                if custom_block_types:
+                    for block_type in custom_block_types:
+                        block_types_filter.append(block_type)
+
+            except ObjectDoesNotExist:
+                logger.error("The site with name %s does not exist", self.site_name)
+            except MultipleObjectsReturned:
+                logger.error(
+                    "The site with name %s returns multiple objects", self.site_name)
+        else:
+            custom_block_types = configuration_helpers.get_value(CUSTOM_BLOCK_TYPES)
+
+            if custom_block_types:
+                    for block_type in custom_block_types:
+                        block_types_filter.append(block_type)
+
+        # filter blocks by types
+        if block_types_filter:
+            block_keys_to_remove = []
+            for block_key in blocks:
+                block_type = blocks.get_xblock_field(block_key, 'category')
+                if block_type not in block_types_filter:
+                    block_keys_to_remove.append(block_key)
+            for block_key in block_keys_to_remove:
+                blocks.remove_block(block_key, keep_descendants=True)
+
+        return blocks
+
+    def get_completed_activities(self, user):
+        """
+        Return blocks that have been completed by the user
+        """
+        return BlockCompletion.get_course_completions(user, self.course_key)
+
+    def get_required_ids(self):
+        """
+        This get the ids for the required activities
+        """
+        if self.site_name is not None:
+            try:
+                site = Site.objects.get(name=self.site_name)
+                return site.microsite.values.get(REQUIRED_IDS, [])
+
+            except ObjectDoesNotExist:
+                logger.error("The site with name %s does not exist", self.site_name)
+            except MultipleObjectsReturned:
+                logger.error(
+                    "The site with name %s returns multiple objects", self.site_name)
+        else:
+            return configuration_helpers.get_value(REQUIRED_IDS, [])
+
+    def is_activity_completed(self, id, activities):
+        """
+        Verify if the id exist for the given activity list
+        """
+        for activity in activities:
+            if id == activity.block_id:
+                return True
+        return False
+
+    def store_report(self, rows):
```

Comment:
Please check the Nico's comment on the Jira issue about configuring the s3 storage according to a microsite setting